### PR TITLE
Update and fix copyright headers

### LINKED ISSUE #628: Add KeePassXC Team Copyright to modified source files
Any source file that has been modified for KeePassXC shall carry the Team's copyright on the top along with the original copyright from KeePassX. Any new files shall only carry the Team's copyright (e.g., SearchWidget.cpp).

Also, add any new files with appropriate copyright to the COPYING file.

### PR DESCRIPTION
**Pull request by TheZ3ro**
Closes #628 

## Description

Intro on copyright for free software projects: 
https://www.softwarefreedom.org/resources/2012/ManagingCopyrightInformation.html
https://opensource.com/law/14/2/copyright-statements-source-files

Note the following points:

> - If N people contribute to a file, are we supposed to have N copyright statements in the file? This doesn't scale over time. Imagine what these files will look like 10 years from now, and fix the problem now.
> - Having author names in a file encourages people to contribute for the wrong reasons. (Note: Contribute for better software instead of contribute for fame and money)
> - Git keeps track of who contributed what changes. It's not necessary to have explicit copyright statements.

> At the Apache Software foundation we discourage the use of author tags in source code. There are various reasons for this, apart from the legal ramifications. Collaborative development is about working on projects as a group and caring for the project as a group. Giving credit is good, and should be done, but in a way that does not allow for false attribution, even by implication. There is no clear line for when to add or remove an author tag. Do you add your name when you change a comment? When you put in a one-line fix? Do you remove other author tags when you refactor the code and it looks 95% different? What do you do about people who go about touching every file, changing just enough to make the virtual author tag quota, so that their name will be everywhere?

---

With those note in minds, what I've done here is pretty simple:
 - Compare with meld or a similar tool the **keepassx** and **keepassxc** repo folder
 - For every `.cpp` file that has difference and has been modified after the fork:
   - If there are substantial changes, add the new copyright notice.
   - If the `.cpp` file comes with an `.h` file, add the new copyright to that as well
   - If there are minor changes don't touch the file. (Useless adding copyright to that simple change like Q_EMIT to emit conversions)
- For every `.cpp` file that has a copyright notice and was added after the fork, maintain the copyright and add *KeePassXC Team* (we can't really remove the copyright without asking the author)
- `.ui` files don't need copyright notice
- For every `CMakeLists.txt`, if the file has some code (it's >10 lines long) add the copyright notice
- For every `.png` file, add the copyright notice to the **COPYING** file
- For every external/library code, add the copyright notice to the **COPYING** file

---

Other then this, I have to report some other things:
- Fixed the **COPYING** file, it was reporting KeePassX information/copyright
- Added a Comment in the **COPYING** file explaining what **KeePassXC Team** means. Some team members' email are missing.
- Removed from the zxcvbn copyright notice in **COPYING** the KeePassXC name (we have no copyright on that)
- Removed from `src/cli/EntropyMeter.cpp` the **zxcvbn** copyright since it was rewritten by me and (mostly by) @louib 
- The following files were reporting `Felix Geyer` copyright but they were added after the fork by different people, I've changed the copyright to `KeePassXC Team`
   -  `src/cli/Extract.cpp`
   -   `src/cli/Merge.cpp`
   -  `src/core/PassphraseGenerator.cpp`
   -  `src/core/PassphraseGenerator.h`
   -  `src/gui/CloneDialog.cpp`
   - `src/gui/CloneDialog.h`
   - `src/http/HttpPasswordGeneratorWidget.cpp`
   - `src/keys/YkChallengeResponseKey.h`
- **All KeePassHTTP** files were reporting the copyright in a different format, changed the format and added `KeePassXC Team` notice
- Copyright for `share/icons/application/16x16/actions/group-empty-trash.png` is missing, like noted [here](https://github.com/keepassxreboot/keepassxc/commit/dea65b637ca2369524b9b1864eff3b2457fb37b7#commitcomment-22471051).

---

I hope I've done a fine work. Waiting for feedback on this. @keepassxreboot/core-developers 

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

**Comment by droidmonkey** [comment]
Wow this was great work! I'll do a thorough review and approve.

**Comment by louib** [comment]
good job @TheZ3ro !!

**Comment by TheZ3ro** [comment]
Uh yeah I missed that. Here we have a small problem.
Apache2.0 license is [compatible only](https://www.gnu.org/licenses/license-list.en.html#apache2) with GNU GPLv3.0 and not with GNU GPLv2.0.
KeePassXC is currently licensed on both v3.0 and v2.0 (at choice).

**Comment by weslly** [comment]
@TheZ3ro I'm no license expert but since KeePassXC doesn't specify a version of GPL and is dual licensed, wouldn't this situation be covered by this line of GPLv2?

> This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; **either version 2 of the License, or (at your option) any later version.**

**Comment by droidmonkey** [comment]
GPLv2 is inherently incompatible with GPLv3. Therefore you must pick and choose the sections that apply in both. Per my reading, section 6 in v2 is the problem. If we decide that v3 is in force for that section its all good. IANL

**Comment by phoerious** [comment]
We should use GPLv3 if you ask me.

**Comment by TheZ3ro** [comment]
I've added the APACHE-2.0 license. 
I would really like a different implementation for base32 (maybe written by us) for the next version of KeePassXC.

File: src/totp/base32.cpp [hunk 1]
```diff
@@ -1,9 +1,8 @@
 // Base32 implementation
+// Source: https://github.com/google/google-authenticator-libpam/blob/master/src/base32.c
 //
 // Copyright 2010 Google Inc.
 // Author: Markus Gutschke
-// Source: https://github.com/google/google-authenticator-libpam/blob/master/src/base32.c
-// Modifications copyright (C) 2017 KeePassXC team <https://keepassxc.org/>
```
Comment (weslly) [inline src/totp/base32.cpp]: @TheZ3ro just a small nitpick, doesn't [apache v2.0 requires a modification notice](https://softwareengineering.stackexchange.com/a/220079)? Most of this function was modified in comparison to the original file

Comment (TheZ3ro) [inline src/totp/base32.cpp]: Yup, fixing this

**Pull request merged**